app/components/message-sender/MessageSender.js

```diff
@@ -0,0 +1,27 @@
+(function () {
+    'use strict';
+
+    var module = angular.module('chatApp.MessageSender', ['chatApp.DataModel']);
+        
+    module.controller('MessageSenderController', function($scope, DataModelService) {
+            $scope.isLoggedIn = function() {
+                return DataModelService.isLoggedIn();
+            };
+            $scope.messageText = '';
+            $scope.postMessage = function() {
```

Comment:
How do you validate the form?
Have you tried to send form instance and check $invalid property?